File: araok/src/main/java/ru/otus/dto/ContentWithContentMediaAndMediaSubtitleDto.java
package ru.otus.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ContentWithContentMediaAndMediaSubtitleDto {
    private ContentDto content;

    private MediaSubtitleDto mediaSubtitle;

    private ContentMediaDto preview;

    private ContentMediaDto video;
}

File: araok-android/src/main/java/ru/otus/data/dto/ContentWithContentMediaAndMediaSubtitleDto.kt
package ru.araok.data.dto

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import ru.araok.entites.ContentWithContentMediaAndMediaSubtitle

@JsonClass(generateAdapter = true)
data class ContentWithContentMediaAndMediaSubtitleDto(
    @Json(name = "content") override val content: ContentDto,
    @Json(name = "mediaSubtitle") override val mediaSubtitle: MediaSubtitleDto,
    @Json(name = "preview") override val preview: ContentMediaDto,
    @Json(name = "video") override val video: ContentMediaDto
): ContentWithContentMediaAndMediaSubtitle
